From db162ffda244901300f3692f96b016d594306b9e Mon Sep 17 00:00:00 2001
From: Stefan Roesch <shr@devkernel.io>
Date: Tue, 5 Dec 2023 10:04:55 -0800
Subject: [PATCH v4 0/2] KSM: support smart-scan feature

This patch series makes two changes:
- Disable the smart-scan feature for ksm01 - ksm04
  This is necessary to make sure that the volatile metrics have the
  expected counts.
- Add a new test for the smart-scan feature
  The new test verifies that the smart-scan feature skips pages when it has
  been enabled for a VMA.

Versions:
- V4:
  - no changes to first patch

  - Changed header documentation
  - Fixed includes
  - Renamed create_memory() to verify_ksm(), orignal verify_ksm() has been
    removed
  - Remove loop in verify_ksm
  - Remove wait at the end
  - Call madvise at the end of the function
  - Remove needs_fork
  
- V3:
  - Disable smart scan for ksm05 and ksm06
  - add ksm07 to runtest/mm
  - change documentation format in header of ksm07.c
  - Move function create_memory_for_smartscan() to ksm07.c
  - Rename create_memory_for_smartscan() to create_memory()
  
- V2:
  - Disable smart_scan in the test setup structure
  - Remove the changes in create_same_memory()
  - Add the new testcase ksm07 for the smart scan test
  

*** BLURB HERE ***

Stefan Roesch (2):
  mem: disable KSM smart scan for ksm tests
  add ksm test for smart-scan feature

 runtest/mm                       |   1 +
 testcases/kernel/mem/.gitignore  |   1 +
 testcases/kernel/mem/ksm/ksm01.c |   2 +
 testcases/kernel/mem/ksm/ksm02.c |   2 +
 testcases/kernel/mem/ksm/ksm03.c |   2 +
 testcases/kernel/mem/ksm/ksm04.c |   2 +
 testcases/kernel/mem/ksm/ksm05.c |   2 +
 testcases/kernel/mem/ksm/ksm06.c |   2 +
 testcases/kernel/mem/ksm/ksm07.c | 113 +++++++++++++++++++++++++++++++
 9 files changed, 127 insertions(+)
 create mode 100644 testcases/kernel/mem/ksm/ksm07.c


base-commit: 8c89ef3d451087ed6e18750bd5eedd10e5ab3d2e
-- 
2.39.3

